### PR TITLE
Remove `offset-converter` run dependency. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Remove `offset-converter` run dependency. NFC

Following up on the change made in #14611, this change forces the
`WasmOffsetConverter` before calling receiveInstantiationResult in both
code paths (instantiateArrayBuffer and instantiateAsync).

With that done (along with #14755) there are no more valid use cases for
to block execution based on the absence of `WasmOffsetConverter`.  By
the time the instance is received it is always defined.

### CODE REVIEW

File: src/preamble.js [hunk 1]
```diff
@@ -1083,15 +1078,20 @@ function createWasm() {
   }
 
   function instantiateArrayBuffer(receiver) {
+#if USE_OFFSET_CONVERTER
+    var bin;
```
Comment (kripken) [inline src/preamble.js:1082]: how about `binary_`, or `savedBinary`?

**Review by kripken — APPROVED** [review]

File: src/preamble.js [hunk 2]
```diff
@@ -1083,15 +1078,20 @@ function createWasm() {
...
-        {{{ runOnMainThread("removeRunDependency('offset-converter');") }}}
-      });
-#endif // USE_OFFSET_CONVERTER
-      return result;
+      bin = binary;
+#endif
+      return WebAssembly.instantiate(binary, info);
+    }).then(function (instance) {
+#if USE_OFFSET_CONVERTER
+      // See comment below in instantiateAsync
```
[the remark this answers is not in the document]
Comment (sbc100) [reply src/preamble.js:1091]: The one on line 1139: `When using the offset converter, we must interpose here.`.

Basically we can't call `receiver` before we have assigned `wasmOffsetConverter` .. for the same reason as on line 1139.   I believe you fixed this in #14611 for the normal `instantiateAsync` but not for the fallback `instantiateArrayBuffer`.

File: src/preamble.js [hunk 3]
```diff
@@ -1083,15 +1078,21 @@ function createWasm() {
...
-        {{{ runOnMainThread("removeRunDependency('offset-converter');") }}}
-      });
-#endif // USE_OFFSET_CONVERTER
-      return result;
+      savedBinary = binary;
+#endif
+      return WebAssembly.instantiate(binary, info);
+    }).then(function (instance) {
+#if USE_OFFSET_CONVERTER
+      // wasmOffsetConverter needs to be assigned because calling receiver
```
Comment (kripken) [inline src/preamble.js:1091]:
```suggestion
      // wasmOffsetConverter needs to be assigned before calling the receiver
```

**Review by kripken — APPROVED** [review]
I see, makes sense. Yes, I think that's right.

**Pull request merged**